velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -57,6 +71,255 @@ void ensureRegexIsCompatible(
   }
 }
 
+// Blocks patterns that contain character class union, intersection, or
+// difference because these are not understood by RE2 and will be parsed as a
+// different pattern than in java.util.regex.
+void ensureRegexIsCompatible(
+    const char* functionName,
+    const VectorPtr& patternVector) {
+  if (!patternVector ||
+      patternVector->encoding() != VectorEncoding::Simple::CONSTANT) {
+    VELOX_USER_FAIL("{} requires a constant pattern.", functionName);
+  }
+  if (patternVector->isNullAt(0)) {
+    return;
+  }
+  const StringView pattern =
+      patternVector->as<ConstantVector<StringView>>()->valueAt(0);
+  // Call the factored out function to check the pattern.
+  checkForCompatiblePattern(
+      std::string(pattern.data(), pattern.size()), functionName);
+}
+class Re2ReplaceConstantPattern final : public exec::VectorFunction {
+ public:
+  /**
```

Comment:
Please, review coding standards and update to comply.

Reply:
Modified class/documentation exposed comments to be triple-slashed and removed obvious comments in recent commit